### ISSUE TITLE
Unable to set NRPEMODE during silent install

### ISSUE THREAD

**Issue by ubergeek42** [mickem/nscp#257]
Setting NRPEMODE to LEGACY(to work with a shinken server) results in the value being overwritten by the ApplyTool function(I think).

Logs show this:

```
MSI (s) (A8:EC) [14:08:15:902]: PROPERTY CHANGE: Adding PackagecodeChanging property. Its value is '1'.
MSI (s) (A8:EC) [14:08:15:902]: Package name extracted from package path: 'NSCP-0.4.4.19-x64.msi'
MSI (s) (A8:EC) [14:08:15:902]: Package to be registered: 'NSCP-0.4.4.19-x64.msi'
MSI (s) (A8:EC) [14:08:15:902]: Note: 1: 2262 2: AdminProperties 3: -2147287038 
MSI (s) (A8:EC) [14:08:15:902]: Machine policy value 'DisableMsi' is 0
MSI (s) (A8:EC) [14:08:15:902]: Machine policy value 'AlwaysInstallElevated' is 0
MSI (s) (A8:EC) [14:08:15:902]: User policy value 'AlwaysInstallElevated' is 0
MSI (s) (A8:EC) [14:08:15:902]: Product installation will be elevated because user is admin and product is being installed per-machine.
MSI (s) (A8:EC) [14:08:15:902]: Running product '{655CAB54-9D54-4F36-8485-6DC5BC0D00AA}' with elevated privileges: Product is assigned.
MSI (s) (A8:EC) [14:08:15:902]: PROPERTY CHANGE: Adding CONF_CAN_CHANGE property. Its value is '1'.
MSI (s) (A8:EC) [14:08:15:902]: PROPERTY CHANGE: Adding CONF_NRPE property. Its value is '1'.
MSI (s) (A8:EC) [14:08:15:902]: PROPERTY CHANGE: Modifying ALLOWED_HOSTS property. Its current value is '127.0.0.1,::1'. Its new value: '163.118.135.13'.
MSI (s) (A8:EC) [14:08:15:902]: PROPERTY CHANGE: Modifying NRPEMODE property. Its current value is 'SAFE'. Its new value: 'LEGACY'.
MSI (s) (A8:EC) [14:08:15:902]: PROPERTY CHANGE: Modifying MONITORING_TOOL property. Its current value is 'GENERIC'. Its new value: 'none'.
MSI (s) (A8:EC) [14:08:15:902]: PROPERTY CHANGE: Modifying CONFIGURATION_TYPE property. Its current value is 'ini://${exe-path}/nsclient.ini'. Its new value: 'ini://${shared-path}/nsclient.ini'.

--- SNIP ---

MSI (s) (A8:AC) [14:08:16:340]: Invoking remote custom action. DLL: C:\WINDOWS\Installer\MSIE9F0.tmp, Entrypoint: ApplyTool
Action start 14:08:16: ApplyTool.
1: Applying monitorin tool config 
MSI (s) (A8!98) [14:08:16:371]: PROPERTY CHANGE: Modifying NSCLIENT_PWD property. Its current value is '$GEN$'. Its new value: 'VD5mgxH9uNhOS4js'.
MSI (s) (A8!98) [14:08:16:371]: PROPERTY CHANGE: Adding CONF_CHECKS property. Its value is '1'.
MSI (s) (A8!98) [14:08:16:386]: PROPERTY CHANGE: Modifying NRPEMODE property. Its current value is 'LEGACY'. Its new value: 'SAFE'.
MSI (s) (A8!98) [14:08:16:386]: PROPERTY CHANGE: Deleting INSTALL_SAMPLE_CONFIG property. Its current value is '0'.
MSI (s) (A8!98) [14:08:16:386]: PROPERTY CHANGE: Deleting GENERATE_SAMPLE_CONFIG property. Its current value is '0'.
MSI (s) (A8:AC) [14:08:16:386]: NOTE: custom action ApplyTool unexpectedly closed the hInstall handle (type MSIHANDLE) provided to it. The custom action should be fixed to not close that handle.
1: Monitoring tool is: none 
MSI (s) (A8:EC) [14:08:16:386]: Doing action: CostFinalize
Action ended 14:08:16: ApplyTool. Return value 1.
MSI (s) (A8:EC) [14:08:16:386]: PROPERTY CHANGE: Adding OutOfDiskSpace property. Its value is '0'.
MSI (s) (A8:EC) [14:08:16:386]: PROPERTY CHANGE: Adding OutOfNoRbDiskSpace property. Its value is '0'.
MSI (s) (A8:EC) [14:08:16:386]: PROPERTY CHANGE: Adding PrimaryVolumeSpaceAvailable property. Its value is '0'.
```

I now see that it's likely due to this else statement here:
https://github.com/mickem/nscp/blob/0.4.4.19/libs/installer_lib/installer_lib.cpp#L192

And I see you've already fixed this in the master branch(here: https://github.com/mickem/nscp/commit/fac1bbf56cb264ef1da49973b426abc403659f40 ). Is there any chance of you cutting a new 4.4.x release with this fix? Or are you aware of a different workaround for this?

**Comment by mickem** [comment]
Just to clearify you need to set MONITORING_TOOL=foo or some other nonsense value to prevent overriding (which in terms is a bug of its own as well).

Sorry, I thought this fix was backported already, will do ASAP.

**Comment by ubergeek42** [comment]
Yes, I discovered that when reading through the code(though I was looking at the master branch). In the logs above you can see it set it to "none".

No worries, I was able to work around around the issue for now. Thanks for the software, it works great!

**Comment by mickem** [comment]
This is fixed in 0.5.1
